### PR TITLE
Add multi-vc configuration for e2e tests

### LINKED ISSUE #412: K8s e2e tests configuration for multi-vc
Currently, we accept environment variables for single VC, we need to change to multi-vc.

### PR DESCRIPTION
**Pull request by abrarshivani**
**What this PR does / why we need it**:
Currently, we accept configuration for only single VC in e2e tests. This PR adds support for multiple VC configuration for e2e tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/vmware/kubernetes/issues/412

**Special notes for your reviewer**:
Internally reviewed here: https://github.com/vmware/kubernetes/pull/418

**Release note**:

```release-note
NONE
```
// cc @divyenpatel @shaominchen

### CODE REVIEW

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (jeffvance) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: why not `const`?

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (jeffvance) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: move `defer` to after the err check?

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (jeffvance) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: no need to check `err` here?

File: test/e2e/storage/vsphere/vsphere_common.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_common.go]: Would be beneficial (less duplicated code perhaps?) to add error handling in this file? IOW handle the cases of a `os.Genenv` error and the case of the variable being empty or not in an expected range?

File: test/e2e/storage/vsphere/vsphere_common.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_common.go]: can all of these vars be made `const`s?

File: test/e2e/storage/vsphere/vsphere_scale.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_scale.go]: The name of the env var now appears in 2 different files. It might be better to define the **name* * of the env vars as const too.

File: test/e2e/storage/vsphere/vsphere_stress.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_stress.go]: ditto...

File: test/e2e/storage/vsphere/vsphere_volume_diskformat.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_volume_diskformat.go]: check err here?

File: test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go]: nit: lowercase "T" in msg

File: test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go]: ditto...

File: test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go
Comment (jeffvance) [inline test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go]: ditto...

File: test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go
Comment (abrarshivani) [reply test/e2e/storage/vsphere/vsphere_volume_vsan_policy.go]: done

File: test/e2e/storage/vsphere/vsphere_common.go
Comment (abrarshivani) [reply test/e2e/storage/vsphere/vsphere_common.go]: This sounds good to me. Made env var const.

File: test/e2e/storage/vsphere/vsphere_common.go
Comment (abrarshivani) [reply test/e2e/storage/vsphere/vsphere_common.go]: Added error handling in this file.

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (abrarshivani) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: Good Catch! Added code to check error.

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (abrarshivani) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: Good Catch! Moved defer

**Comment by divyenpatel** [comment]
@jeffvance pull-kubernetes-verify  is failing with following

```
I0103 22:23:50.153] Verifying verify-flags-underscore.py
I0103 22:23:52.675] Found a flag declared with an _ but which is not explicitly listed as a valid flag name in hack/verify-flags/excluded-flags.txt
I0103 22:23:52.676] Are you certain this flag should not have been declared with an - instead?
I0103 22:23:52.676] Error Parsing VCP_STRESS_ITERATIONS
I0103 22:23:52.680] [0;31mFAILED[0m   verify-flags-underscore.py	2s
I0103 22:23:52.681] ========================
I0103 22:23:52.681] [0;31mFAILED TESTS[0m
I0103 22:23:52.682] ========================
```

Do we need to add any flag into `hack/verify-flags/excluded-flags.txt`? I am not sure why `verify-flags-underscore.py` is not complaining about other parameters with underscore.

**Comment by BaluDontu** [comment]
Verifying verify-flags-underscore.py is failing because it found a flag "VCP_STRESS_ITERATIONS" declared with an _ which is not explicitly listed as a valid flag name in hack/verify-flags/excluded-flags.txt.

The problem is with this specific line
`Expect(err).NotTo(HaveOccurred(), "Error Parsing VCP_STRESS_ITERATIONS")`

where it expects to not have _ as part of NotTo statement. Since this is an environment variable which needs to defined for running vsphere tests, I have explicitly added it to hack/verify-flags/excluded-flags.txt so that verify-flags-underscore.py doesn't complain anymore.

Please refer the latest commit for this change.

**Comment by BaluDontu** [comment]
@jeffvance : Can you provide lgtm if everything looks good ?

**Comment by BaluDontu** [comment]
@jingxu97: Can you please approve this PR. I see you in one of the approvers list for test/e2e/storage/OWNERS

**Comment by BaluDontu** [comment]
@jingxu97 : What are the approvers from sig-cli-maintainers which are part of hack/verify-flags/OWNERS.

**Comment by BaluDontu** [comment]
@janetkuo : Can you please review this PR and provide an approval if everything looks good.

**Comment by jingxu97** [comment]
cc @janetkuo Could you please help approve this PR?

**Review by janetkuo — APPROVED** [review]
Please squash commits

**Comment by BaluDontu** [comment]
@janetkuo : I squashed the commits and rebased the branch. could you please provide your approval.

**Comment by BaluDontu** [comment]
@jingxu97 @jeffvance: Since I had to rebase the PR, could provider your lgtm again ?

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (liggitt) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: is this intentionally global so it can be mutated from other packages? that makes it hard to reason about side effects

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (liggitt) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: why not pass in confFileLocation or look up from env within this method?

File: hack/verify-flags/excluded-flags.txt
Comment (liggitt) [inline hack/verify-flags/excluded-flags.txt]: is this actually a flag?

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (BaluDontu) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: Nope. I have made it as a local variable. Please check the latest commit.

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (BaluDontu) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: I have modified the look up for env in this method only. Please check the latest commit.

File: hack/verify-flags/excluded-flags.txt
Comment (BaluDontu) [reply hack/verify-flags/excluded-flags.txt]: Its not exactly. But make-verify fails without adding this flag. i have explained the reason in comment - https://github.com/kubernetes/kubernetes/pull/57258#issuecomment-356449396

**Comment by BaluDontu** [comment]
@liggitt  @jingxu97 @jeffvance : Can you please verify and provide your approval.

File: hack/verify-flags/excluded-flags.txt
Comment (liggitt) [reply hack/verify-flags/excluded-flags.txt]: thanks

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go
Comment (liggitt) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: thanks

**Pull request merged**
